### PR TITLE
refactor(ngAnimateSwap): remove unused dependency and add tests

### PR DESCRIPTION
**Pull request by gkalpak**
**What kind of change does this PR introduce? (Bug fix, feature, docs update, ...)**
Refactor.
Tests.

**Does this PR introduce a breaking change?**
No.

**Please check if the PR fulfills these requirements**
- [x] The commit message follows our [guidelines](https://github.com/angular/angular.js/blob/master/DEVELOPERS.md#commits)
- [ ] ~~Fix/Feature: [Docs](https://github.com/angular/angular.js/blob/master/DEVELOPERS.md#documentation) have been added/updated~~
- [x] Fix/Feature: Tests have been added; existing tests pass

### CODE REVIEW

**Review by petebacondarwin — APPROVED** [review]
LGTM a couple of minor points

File: test/ngAnimate/ngAnimateSwapSpec.js [hunk 1]
```diff
@@ -132,6 +132,51 @@ describe('ngAnimateSwap', function() {
     expect(two).toBeTruthy();
   }));
 
+  it('should create a new (non-isolate) scope for each inserted clone', inject(function() {
```
[the remark this answers is not in the document]
Comment (Narretz) [reply test/ngAnimate/ngAnimateSwapSpec.js]: Wouldn't an isolate scope be directly derived from the rootScope and make this fail: `expect(scopeOne.$parent).toBe(parentScope)`

**Comment by Narretz** [comment]
LGTM, but the `test` commit should be another `refactor` because you've changed the implementation a bit.

File: test/ngAnimate/ngAnimateSwapSpec.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (petebacondarwin) [reply test/ngAnimate/ngAnimateSwapSpec.js]: No - an isolate scope still has a parent but it doesn't inherit from its parent. So

```
$parent.foo = 'bar';
$isoChild = $parent.$new(true);
assert($isoChild.foo !== 'bar');
assert($isoChild.$parent.foo === 'bar');
```

[the remark this answers is not in the document]
Comment (gkalpak) [reply test/ngAnimate/ngAnimateSwapSpec.js]: Ooops :grin:

**Pull request closed without merging**